README.md

```diff
@@ -1,11 +1,158 @@
+
+## generateFixedPeriods
+
+- `generateFixedPeriods` returns the periods of a specific type in a specific year. This is used, for example in [Data Visualiser](https://github.com/dhis2/data-visualizer-app/pull/2233), to display the list of periods of different types (i.e. monthly, weekly, yearly) in a specific year, according to the user's calendar and locale.
+
+The method takes an `options` object of type `GeneratedPeriodParams` and returns an `Array<FixedPeriod>`.
+
+```ts
+type GeneratedPeriodParams = {
+    year: number
+    periodType: PeriodIdentifier
+    calendar: SupportedCalendar
+    locale?: string
+    startingDay?: number /** 1 is Monday */
+    yearsCount?: number
+}
+```
+- For convenience, `locale` can be passed in the Java-like style (i.e. `ar_SD` rather than `ar-SD`) and it will be converted to the JS-style.
+
+- These ar the supported periodTypes:
+
+```
+const periodIdentifiers = [
+    'DAILY',
+    'WEEKLY',
+    'WEEKLYWED',
+    'WEEKLYTHU',
+    'WEEKLYSAT',
+    'WEEKLYSUN',
+    'BIWEEKLY',
+    'MONTHLY',
+    'BIMONTHLY',
+    'QUARTERLY',
+    'QUARTERLYNOV', // used in Ethiopia
+    'SIXMONTHLY',
+    'SIXMONTHLYAPR',
+    'SIXMONTHLYNOV', // used in Ethiopia
+    'YEARLY',
+    'FYNOV',
+    'FYOCT',
+    'FYJUL',
+    'FYAPR',
+] as const
+```
+
+Although, the library internally is flexible, and can accept, for example, quarterly periods starting any month (i.e. `QUARTERNLYMAY`), support for such periods might not be available in the Backend so should be used carefully. The flexibility, though, means that a new period could be added to the backend and used without changes in the frontend.
+
+
+The result value is of type `FixedPeriod`:
+
+```ts
+type FixedPeriod = {
+    id: string
+    iso?: string
+    name: string
+    startDate: string
+    endDate: string
+}
+```
+
+- `id` and `iso` are the same right now, and return a period identifier, i.e. `2015Q1` (quarter one of the year 2015) or `2015SunW1` (the first week - starting Sunday - of the year 2015). The full list of values are documented in [periodValues](https://github.com/dhis2/multi-calendar-dates/blob/main/features/fixed-periods.ethiopic.feature) in the feature files.
+
+- `startDate` and `endDate` are the start and end dates of the specific period. This is provided for convenience, but the backend currently only makes use of the ID to work out what each period means.
+
+- `name` is the human readable name of the period, according to what we dispaly in DHIS2 Data Visualizer, i.e. `Week 1 - 2014-13-03 - 2015-01-04` for a weekly period or `2015` for a yearly period. 
+
+## getNowInCalendar
+
+- `getNowInCalendar` returns a ZonedDateTime in the calendar for today's date. 
```

Comment:
```suggestion
`getNowInCalendar` returns a [`ZonedDateTime`](https://tc39.es/proposal-temporal/docs/zoneddatetime.html) in the calendar for today's date. 
```

Instead of the link, you could also write the `ZonedDateTime` type here inline like you did for some others 🙂 (also, does this function take no arguments?)